### DISCUSSION TITLE
Icons

### DISCUSSION THREAD

**Discussion by Alex-Wenner-FHR** [h2oai/wave#943] [Ideas] [answered]
This is kind of a 3 part feature request/inquiry

- I would really like the ability to rename icons similarly to how you have the ability to color them (don't know if this is possible because the names come from ui fabricon)
- I would really like to attach actions to icons similar way a button would (```q.args.<icon_name>```)
-  I would really like the ability to add icons to every card. Right now I believe (if I am not mistaken) you can only put icons on specific cards. 

If any of these currently exist please do follow up! I did some digging but didn't find any resources on this. Thanks!

**Comment by mturoci** [comment, marked as answer]
> I would really like the ability to rename icons similarly to how you have the ability to color them (don't know if this is possible because the names come from ui fabricon)

Not sure what you mean here, how would you like the API to look like? What is the motivation behind such renaming?

> I would really like to attach actions to icons similar way a button would (q.args.<icon_name>)

There is an `icon` prop in button component that has this behavior.

> I would really like the ability to add icons to every card. Right now I believe (if I am not mistaken) you can only put icons on specific cards.

We try to add icon support to components we think might need them - buttons, menu items etc. Can you maybe point out other components that you feel like should support them too?

**Reply by Alex-Wenner-FHR** [reply]
Yes! 
**Icon Naming:**
Naming
So for the naming of the icons it would be cool to have label and name capabilities as other componets. The reason for the name would be say, placing an icon at the top of a component and doing something like:
```
if q.args.<icon_name> : 
       del q.page['card']
```

Labeling
My first column in my table is static and only is made up of an icon. However when I group by this Icon in the table I see its name from uifabricicon 
![image](https://user-images.githubusercontent.com/78372056/128083139-041be8ae-d0f6-4254-bfc6-2f54f6604415.png)


**Icon availability per component:**
I think it would be good to have them on a majority of components. Ex: If I have a form card and want to use an icon as a submit button, rather than a button with an icon in it. Or a form card with an icon 'X' in the top that allows me to close the card or clear it (maybe comes back to q.args.<icon_name>)

let me know if this doesn't make sense!

**Reply by mturoci** [reply]
> My first column in my table is static and only is made up of an icon. However when I group by this Icon in the table I see its name from uifabricicon

Agree, but this is something we need to address at a table level

> Ex: If I have a form card and want to use an icon as a submit button, rather than a button with an icon in it. Or a form card with an icon 'X' in the top that allows me to close the card or clear it (maybe comes back to q.args.<icon_name>)

Makes sense, I would however suggest adding something like `iconOnly` prop to a button which would render [icon button](https://developer.microsoft.com/en-us/fluentui#/controls/web/button) like here - cc @lo5

**Reply by Alex-Wenner-FHR** [reply]
@mturoci any thoughts on a temp workaround for the table naming? But awesome sounds like both of these proposed solutions would be what I am looking for!

**Reply by mturoci** [reply]
Unfortunately no workarounds as for now.

**Reply by lo5** [reply]
@mturoci Good idea. Instead of introducing `icon_only`, maybe we can render `IconButton` if the `label` is empty. The behavior would be similar to how we currently switch between the regular and compound buttons based on the availability of a `caption`.

**Reply by mturoci** [reply]
Makes sense @lo5, will do.
